crates/ast/src/util.rs

```diff
@@ -53,3 +56,79 @@ pub fn u256_as_push(value: U256) -> Opcode {
         _ => unreachable!(),
     }
 }
+
+type Selector = (FixedBytes<4>, FixedBytes<4>);
+
+pub fn compute_selector(
+    func: SolFunction,
+    err: SolError,
+) -> Option<(FixedBytes<4>, FixedBytes<4>)> {
+    let build_signature = |name: &Spanned<&str>, args: &Box<[Spanned<DynSolType>]>| -> Vec<u8> {
+        let arg_types: Vec<_> = args
+            .iter()
+            .map(|arg| match arg.0 {
+                DynSolType::Address => "address",
+                DynSolType::Uint(_) => "uint256",
+                DynSolType::String => "string",
+                _ => panic!("Unsupported type: {:?}", arg.0),
```

Comment:
You don't support all types. I think Alloy might have a built-in way, try `.to_string()`